storage/remote/queue_manager.go

```diff
@@ -302,11 +341,83 @@ func NewQueueManager(reg prometheus.Registerer, metrics *queueManagerMetrics, lo
 	}
 
 	t.watcher = wal.NewWatcher(reg, wal.NewWatcherMetrics(reg), logger, client.Name(), t, walDir)
+	if t.cfg.SendMetadata {
+		t.metadataWatcher = scrape.NewMetadataWatcher(logger, sm, client.Name(), t, t.cfg.MetadataSendInterval)
+	}
 	t.shards = t.newShards()
 
 	return t
 }
 
+// AppendMetadata queues metadata to be sent to the remote storage.
+func (t *QueueManager) AppendMetadata(metadataList []scrape.MetricMetadata) bool {
```

Comment:
Super nit: I personally very much dislike variable names `fooType`.  Just call it `metadata`.